plugin/core/views.py

```diff
@@ -100,15 +100,26 @@ def render_text_change(change: sublime.TextChange) -> Dict[str, Any]:
 
 
 def did_change_text_document_params(view: sublime.View,
-                                    changes: Optional[Iterable[sublime.TextChange]] = None) -> Dict[str, Any]:
+                                    changes: Optional[Sequence[sublime.TextChange]] = None) -> Dict[str, Any]:
     content_changes = []  # type: List[Dict[str, Any]]
     result = {"textDocument": versioned_text_document_identifier(view), "contentChanges": content_changes}
     if changes is None:
         # TextDocumentSyncKindFull
         content_changes.append({"text": entire_content(view)})
-    else:
+    elif len(changes) > 0:
         # TextDocumentSyncKindIncremental
-        for change in changes:
+        stitched_changes = []
+        current = changes[0]
+        for i in range(1, len(changes)):
+            candidate = changes[i]
+            length = len(current.str)
+            if current.b.pt + length == candidate.a.pt and candidate.a.pt == candidate.b.pt:
+                current.str += candidate.str
```

Comment:
Don't we need to update `current.b.pt` after doing that?

Reply:
No, we don't update `current.b` :) Convince yourself that this is correct.

Reply:
Simplifying the TextChange object let's denote it by `(a, b, str)` where `a` and `b` are sublime unicode codepoint offsets into the buffer. If the user types two characters then it'll look like this:

(0, 0, 'a')
(1, 1, 'b')

Stitching this together we should get

(0, 0, 'ab')

Reply:
Oh, I was interpreting them as ranges I guess. :)

Reply:
Hmm, perhaps my code isn't entirely correct after all

```
(0, 10, 'hel')  # replace (0, 10) with 'hel'
(10, 10, 'l')  # insert 'l'
(11, 11, 'o')    # insert 'o'
```

Reply:
I guess the only requirement is

```
current.b.pt == candidate.a.pt and candidate.a.pt == candidate.b.pt
```

Reply:
So you've integrated with this bug?

Reply:
It's not a bug, it just won't stitch it together.

Reply:
So the final changes will look like

```
(0, 10, 'hel')
(10, 10, 'lo')
```
which is OK.

Reply:
Ideally it would look like `(0, 10, 'hello')` though

Reply:
OK, I thought you wanted to change to a different condition anyway. But you've removed that part of your comment so maybe not. :)